libexec/cli/start.exec

```diff
@@ -0,0 +1,181 @@
+            export SINGULARITY_SHELL
+            shift
+        ;;
+        -u|--user)
+            SINGULARITY_NOSUID=1
+            export SINGULARITY_NOSUID
+            shift
+        ;;
+        -w|--writable)
+            shift
+            SINGULARITY_WRITABLE=1
+            export SINGULARITY_WRITABLE
+        ;;
+        -H|--home)
+            shift
+            SINGULARITY_HOME="$1"
+            export SINGULARITY_HOME
+            shift
+        ;;
+        -W|--wdir|--workdir|--workingdir)
+            shift
+            SINGULARITY_WORKDIR="$1"
+            export SINGULARITY_WORKDIR
+            shift
+        ;;
+        -S|--scratchdir|--scratch-dir|--scratch)
+            shift
+            SINGULARITY_SCRATCHDIR="$1,${SINGULARITY_SCRATCHDIR:-}"
+            export SINGULARITY_SCRATCHDIR
+            shift
+        ;;
+        -B|--bind)
+            shift
+            SINGULARITY_BINDPATH="${SINGULARITY_BINDPATH:-},${1:-}"
+            export SINGULARITY_BINDPATH
+            shift
+        ;;
+        -c|--contain)
+            shift
+            SINGULARITY_CONTAIN=1
+            export SINGULARITY_CONTAIN
+        ;;
+        -e|--cleanenv)
+            shift
+            SINGULARITY_CLEANENV=1
+            export SINGULARITY_CLEANENV
+        ;;
+        --pwd)
+            shift
+            SINGULARITY_TARGET_PWD="$1"
+            export SINGULARITY_TARGET_PWD
+            shift
+        ;;
+        -n|--nv)
+            shift
+            for i in `ldconfig -p | grep -E "/libnv|/libcuda|/libEGL|/libGL|/libnvcu|/libvdpau|/libOpenCL|/libOpenGL"`; do
+                if [ -f "$i" ]; then
+                    message 2 "Found NV library: $i\n"
+                    if [ -z "${SINGULARITY_CONTAINLIBS:-}" ]; then
+                        SINGULARITY_CONTAINLIBS="$i"
+                    else
+                        SINGULARITY_CONTAINLIBS="$SINGULARITY_CONTAINLIBS,$i"
+                    fi
+                fi
+            done
+            if [ -z "${SINGULARITY_CONTAINLIBS:-}" ]; then
+                message WARN "Could not find any Nvidia libraries on this host!\n";
+            else
+                export SINGULARITY_CONTAINLIBS
+            fi
+            if NVIDIA_SMI=`which nvidia-smi`; then
+                if [ -n "${SINGULARITY_BINDPATH:-}" ]; then
+                    SINGULARITY_BINDPATH="${SINGULARITY_BINDPATH},${NVIDIA_SMI}"
+                else
+                    SINGULARITY_BINDPATH="${NVIDIA_SMI}"
+                fi
+                export SINGULARITY_BINDPATH
+            else
+                message WARN "Could not find the Nvidia SMI binary to bind into container\n"
+            fi
+        ;;
+        -*)
+            message ERROR "Unknown option: ${1:-}\n"
+            exit 1
+        ;;
+        *)
+            break;
+        ;;
+    esac
+done
+
+
+if [ -z "${SINGULARITY_IMAGE:-}" ]; then
+    SINGULARITY_IMAGE="${1:-}"
+    export SINGULARITY_IMAGE
+    shift
+fi
+
```

Comment:
Too bad we don't have the bash equivalent of haikunator to come up with funny names, if the user doesn't provide one :)